app/products/calls/state/actions.ts

```diff
@@ -349,6 +350,17 @@ export const newCurrentCall = (serverUrl: string, channelId: string, myUserId: s
 };
 
 export const myselfLeftCall = () => {
+    // Set the previous call for actions that need to know (e.g., handleHostRemoved)
+    const currentCall = getCurrentCall();
+    if (currentCall) {
+        setPreviousCall({
+            serverUrl: currentCall.serverUrl,
+            channelId: currentCall.channelId,
+            mySessionId: currentCall.mySessionId,
+        });
+    } else {
+        setPreviousCall(null);
```

Comment:
is this the only use case to set the previous call as null? just double checking

Reply:
This is a typescript issue -- in this case `currentCall` won't be null, but its type can be. But I can see why it might be confusing, so I removed it.
We aren't setting `previousCall` to null anywhere else because it never will be null. You either have a previous call once and then you'll always have it, or you never had a call (null at the start).